### PR TITLE
Some simplify, restructure and some codes

### PR DESCRIPTION
**Pull request by abnsy**

### CODE REVIEW

File: flaml/searcher/flow2.py [hunk 1]
```diff
@@ -364,34 +359,37 @@ def on_trial_complete(
...
+                self.best_obj = obj
+                config = self._configs[trial_id][0]
+                if self.best_config != config:
+                    self._extracted_from_on_trial_result_11(config, result)
+        cost = result.get(self.cost_attr)
+        # record the cost in case it is pruned and cost info is lost
+        self._trial_cost[trial_id] = cost
+
+    # TODO Rename this here and in `on_trial_result`
+    def _extracted_from_on_trial_result_11(self, config, result):
```
Comment (sonichi) [inline flaml/searcher/flow2.py:381]:
```suggestion
    def _update_best_config(self, config, result):
```

File: flaml/searcher/flow2.py [hunk 2]
```diff
@@ -451,38 +440,34 @@ def suggest(self, trial_id: str) -> Optional[Dict]:
...
             self._init_phase = False
-            if self.step >= self.step_lower_bound:
-                # decrease step size
-                self._oldK = self._K if self._K else self._iter_best_config
-                self._K = self.trial_count_proposed + 1
-                self.step *= np.sqrt(self._oldK / self._K)
-            else:
+            if self.step < self.step_lower_bound:
                 return None
+                # decrease step size
```
Comment (sonichi) [inline flaml/searcher/flow2.py:467]:
```suggestion
            # decrease step size
```

**Review by sonichi — CHANGES_REQUESTED** [review]
I didn't review files under nlp/. I'll request @liususan091219 to review it.

**Review by liususan091219 — COMMENTED** [review]
Since flaml/nlp/result_analysis/azure_utils.py and flaml/nlp/result_analysis/wandb_utils.py will be deleted in the next version, please keep these two files unchanged.

**Comment by sonichi** [comment]
merged in #313

**Pull request closed without merging**